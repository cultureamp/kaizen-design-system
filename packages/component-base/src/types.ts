export type OverrideClassName<T> = Omit<T, "className"> & {
  classNameOverride?: string
}

export type DataAttributes = { [key: `data-${string}`]: string | undefined }
