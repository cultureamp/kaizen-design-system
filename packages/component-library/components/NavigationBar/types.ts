import { default as React, ReactElement } from "react"

export type Navigation = {
  primary?: NavigationItem[]
  secondary?: NavigationItem[]
  final?: NavigationItem[]
}

export type NavigationItem = ReactElement<LinkProps> | ReactElement<MenuProps>

export type LinkProps = {
  icon?: React.SVGAttributes<SVGSymbolElement>
  text?: string
  iconOnly?: boolean
  href: string
  active?: boolean
  id?: string
  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void
  target?: "_self" | "_blank"
  hasMenu?: boolean
  badge?: {
    kind: "new" | "notification"
    text: string
  }
  section?: string
}

export type MenuProps = {
  active?: boolean
  header?: React.ReactElement<any>
  items: Array<MenuItem | MenuGroup>
  automationId?: string
  heading: string
  mobileEnabled?: boolean
  section?: string
}

export type MenuItem = {
  label: string
  url: string
  active?: boolean
  method?: "get" | "post" | "put" | "delete"
}

export type MenuGroup = {
  title: string
  items: MenuItem[]
}
