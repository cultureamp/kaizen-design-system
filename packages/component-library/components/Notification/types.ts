import { NotificationType } from "./components/GenericNotification"

type Modify<T, R> = Omit<T, keyof R> & R

export type ToastNotification = {
  id: string
  type: NotificationType
  title: string
  message: React.ReactNode
  autohide: boolean
  autohideDelay?: "short" | "long"
  onHide?: () => void
  automationId?: string
  persistent?: boolean
}

export type ToastNotificationWithOptionals = Modify<
  ToastNotification,
  {
    id?: string
    autohide?: boolean
  }
>

export type AddToastNotification = (
  notification: ToastNotificationWithOptionals
) => void

export type RemoveToastNotification = (notificationId: string) => void

export type ClearToastNotifications = () => void
