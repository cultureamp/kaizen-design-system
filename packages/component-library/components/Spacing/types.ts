/**
 * Spacing is in grid units (24px grid)
 */
export type Spacing = {
  /**
   * padding
   */
  p?: GridFractions
  /**
   * padding top
   */
  pt?: GridFractions
  /**
   * padding right
   */
  pr?: GridFractions
  /**
   * padding bottom
   */
  pb?: GridFractions
  /**
   * padding left
   */
  pl?: GridFractions
  /**
   * padding about the x-axis - padding left and right
   */
  px?: GridFractions
  /**
   * padding about the y-axis - padding top and bottom
   */
  py?: GridFractions
  /**
   * margin
   */
  m?: GridFractions
  /**
   * margin top
   */
  mt?: GridFractions
  /**
   * margin right
   */
  mr?: GridFractions
  /**
   * margin bottom
   */
  mb?: GridFractions
  /**
   * margin left
   */
  ml?: GridFractions
  /**
   * margin about the x-axis - padding left and right
   */
  mx?: GridFractions
  /**
   * margin about the y-axis - padding top and bottom
   */
  my?: GridFractions
}

export type GridFractions =
  | 0
  | 0.25
  | 0.5
  | 0.75
  | 1
  | 1.25
  | 1.5
  | 1.75
  | 2
  | 2.5
  | 3
  | 3.5
  | 4
