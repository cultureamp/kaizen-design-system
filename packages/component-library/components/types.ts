import { AriaAttributes } from "react"

export type Safelist = AriaAttributes & {
  id?: string
  itemProp?: string
  itemScope?: string
  itemType?: string
  itemID?: string
  itemRef?: string
  tabIndex?: string
  title?: string
  role?: string
}
// @TODO add data attributes
// @TODO enforce data-test-id (check name)?

export type AllowedTags = {
  pre: HTMLPreElement
  p: HTMLParagraphElement
  a: HTMLAnchorElement
  div: HTMLDivElement
  span: HTMLSpanElement
  h1: HTMLHeadingElement
  h2: HTMLHeadingElement
  h3: HTMLHeadingElement
  h4: HTMLHeadingElement
  h5: HTMLHeadingElement
  h6: HTMLHeadingElement
}

/**
 * Scale refers to difference in size of components within the same component family
 */
export type Sizing = "xxs" | "xs" | "sm" | "md" | "lg" | "xl" | "xxl"

export type GridFractions = "1/8" | "1/4" | "1/2" | "0" | "1" | "2" | "3" | "4"

export type Breakpoint = {
  mobile: "0"
  tablet: "768"
  desktop: "1024"
}

export type Scale =
  | {
      [key in keyof Breakpoint]?: Sizing
    }
  | Sizing

export type ResponsiveSpacing =
  | {
      [key in keyof Breakpoint]?: GridFractions
    }
  | GridFractions

export type Padding = ResponsiveSpacing
export type Margin = ResponsiveSpacing

export type Spacing = {
  p?: ResponsiveSpacing
  pt?: GridFractions
  pr?: GridFractions
  pb?: GridFractions
  pl?: GridFractions
  px?: GridFractions
  py?: GridFractions
  m?: ResponsiveSpacing
  mt?: GridFractions
  mr?: GridFractions
  mb?: GridFractions
  ml?: GridFractions
  mx?: GridFractions
  my?: GridFractions
}
