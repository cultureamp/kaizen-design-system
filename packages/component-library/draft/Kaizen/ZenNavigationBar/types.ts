import { default as React, ReactElement } from "react"

export type Navigation = {
  primary?: NavigationItem[]
  secondary?: NavigationItem[]
  final?: NavigationItem[]
}

export type NavigationChange = (
  event: React.MouseEvent<HTMLAnchorElement>
) => void

export type NavigationItem = ReactElement<LinkProps> | ReactElement<MenuProps>

export type LinkProps = {
  icon?: React.SVGAttributes<SVGSymbolElement>
  text?: string
  iconOnly?: boolean
  href: string
  active?: boolean
  id?: string
  target?: "_self" | "_blank"
  hasMenu?: boolean
  onClick?: NavigationChange
  menuOpen?: boolean
  badge?: {
    kind: "new" | "notification"
    text: string
  }
  opaque?: boolean
  small?: boolean
  content?: boolean
}

export type MenuProps = {
  header?: React.ReactElement<any>
  items: Array<MenuItemProps | MenuGroupProps>
  automationId?: string
  heading: string
  mobileEnabled?: boolean
  active?: boolean
  icon?: React.SVGAttributes<SVGSymbolElement>
  opaque?: boolean
  small?: boolean
}

export type MenuItemProps = {
  label: string
  url: string
  method?: "get" | "post" | "put" | "delete"
  showArrowIcon?: boolean
  active?: boolean
  onClick?: NavigationChange
}

export type MenuGroupProps = {
  title: string
  items: MenuItemProps[]
  offCanvas?: boolean
  first?: boolean
}
