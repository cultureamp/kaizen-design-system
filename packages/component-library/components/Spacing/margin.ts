import { GridFractions, ResponsiveSpacing, Spacing } from "../types"
import { convertFractionToString } from "./util"

const styles = require("./Margin.module.scss")

export const mt = (unit: GridFractions): string[] => [
  styles[`mt-${convertFractionToString(unit)}`],
]

export const mr = (unit: GridFractions): string[] => [
  styles[`mr-${convertFractionToString(unit)}`],
]

export const mb = (unit: GridFractions): string[] => [
  styles[`mb-${convertFractionToString(unit)}`],
]

export const ml = (unit: GridFractions): string[] => [
  styles[`ml-${convertFractionToString(unit)}`],
]

export const mx = (unit: GridFractions): string[] => [
  styles[`mx-${convertFractionToString(unit)}`],
]

export const my = (unit: GridFractions): string[] => [
  styles[`my-${convertFractionToString(unit)}`],
]

export const m = (unit: ResponsiveSpacing): string[] => {
  const classes: string[] = []
  if (typeof unit === "string") {
    classes.push(styles[`m-${convertFractionToString(unit)}`])
  } else {
    Object.keys(unit).forEach(key => {
      classes.push(styles[`m-${convertFractionToString(unit[key])}--${key}`])
    })
  }

  return classes
}

export const responsiveMarginClasses = ({
  m: margin,
  mt: marginTop,
  mr: marginRight,
  mb: marginBottom,
  ml: marginLeft,
  mx: marginXAxis,
  my: marginYAxis,
}: Spacing): string[] => {
  const classes: string[] = []

  if (margin !== undefined) classes.push(...m(margin))
  if (marginTop !== undefined) classes.push(...mt(marginTop))
  if (marginRight !== undefined) classes.push(...mr(marginRight))
  if (marginBottom !== undefined) classes.push(...mb(marginBottom))
  if (marginLeft !== undefined) classes.push(...ml(marginLeft))
  if (marginXAxis !== undefined) classes.push(...mx(marginXAxis))
  if (marginYAxis !== undefined) classes.push(...my(marginYAxis))

  // if no padding defined on any param, default to 0
  if (
    margin === undefined &&
    marginTop === undefined &&
    marginRight === undefined &&
    marginBottom === undefined &&
    marginLeft === undefined &&
    marginXAxis === undefined &&
    marginYAxis === undefined
  ) {
    classes.push(...m("0"))
  }

  return classes
}
