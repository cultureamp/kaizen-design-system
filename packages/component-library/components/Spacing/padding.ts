import { GridFractions, Spacing } from "./types"
import { convertFractionToString } from "./util"

import styles from "./Padding.module.scss"

export const pt = (unit: GridFractions): string[] => [
  styles[`pt-${convertFractionToString(unit)}`],
]

export const pr = (unit: GridFractions): string[] => [
  styles[`pr-${convertFractionToString(unit)}`],
]

export const pb = (unit: GridFractions): string[] => [
  styles[`pb-${convertFractionToString(unit)}`],
]

export const pl = (unit: GridFractions): string[] => [
  styles[`pl-${convertFractionToString(unit)}`],
]

export const px = (unit: GridFractions): string[] => [
  styles[`pl-${convertFractionToString(unit)}`],
  styles[`pr-${convertFractionToString(unit)}`],
]

export const py = (unit: GridFractions): string[] => [
  styles[`pt-${convertFractionToString(unit)}`],
  styles[`pb-${convertFractionToString(unit)}`],
]

export const p = (unit: GridFractions): string[] => {
  const classes: string[] = []
  if (typeof unit === "number") {
    classes.push(styles[`p-${convertFractionToString(unit)}`])
  } else {
    Object.keys(unit).forEach(key => {
      classes.push(styles[`p-${convertFractionToString(unit[key])}--${key}`])
    })
  }

  return classes
}

export const paddingClasses = ({
  p: padding,
  pt: paddingTop,
  pr: paddingRight,
  pb: paddingBottom,
  pl: paddingLeft,
  px: paddingXAxis,
  py: paddingYAxis,
  rtl = false,
}: { rtl?: boolean } & Spacing): string[] => {
  const classes: string[] = []

  if (padding !== undefined) classes.push(...p(padding))
  if (paddingTop !== undefined) classes.push(...pt(paddingTop))
  if (paddingBottom !== undefined) classes.push(...pb(paddingBottom))
  if (paddingXAxis !== undefined) classes.push(...px(paddingXAxis))
  if (paddingYAxis !== undefined) classes.push(...py(paddingYAxis))

  if (rtl) {
    // for RTL languages, swap the left and right padding
    if (paddingRight !== undefined) classes.push(...pl(paddingRight))
    if (paddingLeft !== undefined) classes.push(...pr(paddingLeft))
  } else {
    if (paddingRight !== undefined) classes.push(...pr(paddingRight))
    if (paddingLeft !== undefined) classes.push(...pl(paddingLeft))
  }

  // if no padding defined on any param, default to 0
  if (
    padding === undefined &&
    paddingTop === undefined &&
    paddingRight === undefined &&
    paddingBottom === undefined &&
    paddingLeft === undefined &&
    paddingXAxis === undefined &&
    paddingYAxis === undefined
  ) {
    classes.push(...p(0))
  }
  return classes
}
