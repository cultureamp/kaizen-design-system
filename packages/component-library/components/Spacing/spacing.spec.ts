import { GridFractions } from "../types"
import { mb, ml, mr, mt, mx, my, responsiveMarginClasses } from "./margin"
import { pb, pl, pr, pt, px, py, responsivePaddingClasses } from "./padding"

type TestObject = {
  unit: GridFractions
  translation: string
}

const allowableUnits: TestObject[] = [
  { unit: "1/8", translation: "eighth" },
  { unit: "1/4", translation: "quarter" },
  { unit: "1/2", translation: "half" },
  { unit: "0", translation: "0" },
  { unit: "1", translation: "1" },
  { unit: "2", translation: "2" },
  { unit: "3", translation: "3" },
  { unit: "4", translation: "4" },
]

describe("padding", () => {
  describe("pt", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding top for the ${unit} unit`, () => {
        expect(pt(unit)).toMatchSnapshot()
      })
    })
  })
  describe("pr", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding right for the ${unit} unit`, () => {
        expect(pr(unit)).toMatchSnapshot()
      })
    })
  })
  describe("pb", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding bottom for the ${unit} unit`, () => {
        expect(pb(unit)).toMatchSnapshot()
      })
    })
  })
  describe("pl", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding left for the ${unit} unit`, () => {
        expect(pl(unit)).toMatchSnapshot()
      })
    })
  })
  describe("px", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding x axis for the ${unit} unit`, () => {
        expect(px(unit)).toMatchSnapshot()
      })
    })
  })
  describe("py", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for padding y axis for the ${unit} unit`, () => {
        expect(py(unit)).toMatchSnapshot()
      })
    })
  })
  describe("responsivePaddingClasses", () => {
    it(`generates a single padding class`, () => {
      expect(responsivePaddingClasses({ p: "1/4" })).toContain(`p-quarter`)
      expect(responsivePaddingClasses({ p: "1/4" })).toHaveLength(1)
    })
    it(`generates multiple padding classes`, () => {
      const classes = responsivePaddingClasses({ px: "1/4", pt: "1/8" })
      expect(classes).toContain("px-quarter")
      expect(classes).toContain("pt-eighth")
      expect(classes).toHaveLength(2)
    })
  })
})

describe("margin", () => {
  describe("mt", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin top for the ${unit} unit`, () => {
        expect(mt(unit)).toMatchSnapshot()
      })
    })
  })
  describe("mr", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin right for the ${unit} unit`, () => {
        expect(mr(unit)).toMatchSnapshot()
      })
    })
  })
  describe("mb", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin bottom for the ${unit} unit`, () => {
        expect(mb(unit)).toMatchSnapshot()
      })
    })
  })
  describe("ml", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin left for the ${unit} unit`, () => {
        expect(ml(unit)).toMatchSnapshot()
      })
    })
  })
  describe("mx", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin x axis for the ${unit} unit`, () => {
        expect(mx(unit)).toMatchSnapshot()
      })
    })
  })
  describe("my", () => {
    allowableUnits.forEach(({ unit, translation }) => {
      it(`Generates the correct classes for margin y axis for the ${unit} unit`, () => {
        expect(my(unit)).toMatchSnapshot()
      })
    })
  })
  describe("responsiveMarginClasses", () => {
    it(`generates a single margin class`, () => {
      expect(responsiveMarginClasses({ m: "1/4" })).toContain(`m-quarter`)
      expect(responsiveMarginClasses({ m: "1/4" })).toHaveLength(1)
    })
    it(`generates multimle margin classes`, () => {
      const classes = responsiveMarginClasses({ mx: "1/4", mt: "1/8" })
      expect(classes).toContain("mx-quarter")
      expect(classes).toContain("mt-eighth")
      expect(classes).toHaveLength(2)
    })
  })
})
