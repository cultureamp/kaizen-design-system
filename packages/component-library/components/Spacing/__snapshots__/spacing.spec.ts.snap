// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`margin mb Generates the correct classes for margin bottom for the 0 unit 1`] = `
Array [
  "mb-0",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 1 unit 1`] = `
Array [
  "mb-1",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 1/2 unit 1`] = `
Array [
  "mb-half",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 1/4 unit 1`] = `
Array [
  "mb-quarter",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 1/8 unit 1`] = `
Array [
  "mb-eighth",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 2 unit 1`] = `
Array [
  "mb-2",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 3 unit 1`] = `
Array [
  "mb-3",
]
`;

exports[`margin mb Generates the correct classes for margin bottom for the 4 unit 1`] = `
Array [
  "mb-4",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 0 unit 1`] = `
Array [
  "ml-0",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 1 unit 1`] = `
Array [
  "ml-1",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 1/2 unit 1`] = `
Array [
  "ml-half",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 1/4 unit 1`] = `
Array [
  "ml-quarter",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 1/8 unit 1`] = `
Array [
  "ml-eighth",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 2 unit 1`] = `
Array [
  "ml-2",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 3 unit 1`] = `
Array [
  "ml-3",
]
`;

exports[`margin ml Generates the correct classes for margin left for the 4 unit 1`] = `
Array [
  "ml-4",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 0 unit 1`] = `
Array [
  "mr-0",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 1 unit 1`] = `
Array [
  "mr-1",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 1/2 unit 1`] = `
Array [
  "mr-half",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 1/4 unit 1`] = `
Array [
  "mr-quarter",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 1/8 unit 1`] = `
Array [
  "mr-eighth",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 2 unit 1`] = `
Array [
  "mr-2",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 3 unit 1`] = `
Array [
  "mr-3",
]
`;

exports[`margin mr Generates the correct classes for margin right for the 4 unit 1`] = `
Array [
  "mr-4",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 0 unit 1`] = `
Array [
  "mt-0",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 1 unit 1`] = `
Array [
  "mt-1",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 1/2 unit 1`] = `
Array [
  "mt-half",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 1/4 unit 1`] = `
Array [
  "mt-quarter",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 1/8 unit 1`] = `
Array [
  "mt-eighth",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 2 unit 1`] = `
Array [
  "mt-2",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 3 unit 1`] = `
Array [
  "mt-3",
]
`;

exports[`margin mt Generates the correct classes for margin top for the 4 unit 1`] = `
Array [
  "mt-4",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 0 unit 1`] = `
Array [
  "mx-0",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 1 unit 1`] = `
Array [
  "mx-1",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 1/2 unit 1`] = `
Array [
  "mx-half",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 1/4 unit 1`] = `
Array [
  "mx-quarter",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 1/8 unit 1`] = `
Array [
  "mx-eighth",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 2 unit 1`] = `
Array [
  "mx-2",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 3 unit 1`] = `
Array [
  "mx-3",
]
`;

exports[`margin mx Generates the correct classes for margin x axis for the 4 unit 1`] = `
Array [
  "mx-4",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 0 unit 1`] = `
Array [
  "my-0",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 1 unit 1`] = `
Array [
  "my-1",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 1/2 unit 1`] = `
Array [
  "my-half",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 1/4 unit 1`] = `
Array [
  "my-quarter",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 1/8 unit 1`] = `
Array [
  "my-eighth",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 2 unit 1`] = `
Array [
  "my-2",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 3 unit 1`] = `
Array [
  "my-3",
]
`;

exports[`margin my Generates the correct classes for margin y axis for the 4 unit 1`] = `
Array [
  "my-4",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 0 unit 1`] = `
Array [
  "pb-0",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 1 unit 1`] = `
Array [
  "pb-1",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 1/2 unit 1`] = `
Array [
  "pb-half",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 1/4 unit 1`] = `
Array [
  "pb-quarter",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 1/8 unit 1`] = `
Array [
  "pb-eighth",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 2 unit 1`] = `
Array [
  "pb-2",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 3 unit 1`] = `
Array [
  "pb-3",
]
`;

exports[`padding pb Generates the correct classes for padding bottom for the 4 unit 1`] = `
Array [
  "pb-4",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 0 unit 1`] = `
Array [
  "pl-0",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 1 unit 1`] = `
Array [
  "pl-1",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 1/2 unit 1`] = `
Array [
  "pl-half",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 1/4 unit 1`] = `
Array [
  "pl-quarter",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 1/8 unit 1`] = `
Array [
  "pl-eighth",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 2 unit 1`] = `
Array [
  "pl-2",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 3 unit 1`] = `
Array [
  "pl-3",
]
`;

exports[`padding pl Generates the correct classes for padding left for the 4 unit 1`] = `
Array [
  "pl-4",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 0 unit 1`] = `
Array [
  "pr-0",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 1 unit 1`] = `
Array [
  "pr-1",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 1/2 unit 1`] = `
Array [
  "pr-half",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 1/4 unit 1`] = `
Array [
  "pr-quarter",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 1/8 unit 1`] = `
Array [
  "pr-eighth",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 2 unit 1`] = `
Array [
  "pr-2",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 3 unit 1`] = `
Array [
  "pr-3",
]
`;

exports[`padding pr Generates the correct classes for padding right for the 4 unit 1`] = `
Array [
  "pr-4",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 0 unit 1`] = `
Array [
  "pt-0",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 1 unit 1`] = `
Array [
  "pt-1",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 1/2 unit 1`] = `
Array [
  "pt-half",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 1/4 unit 1`] = `
Array [
  "pt-quarter",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 1/8 unit 1`] = `
Array [
  "pt-eighth",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 2 unit 1`] = `
Array [
  "pt-2",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 3 unit 1`] = `
Array [
  "pt-3",
]
`;

exports[`padding pt Generates the correct classes for padding top for the 4 unit 1`] = `
Array [
  "pt-4",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 0 unit 1`] = `
Array [
  "px-0",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 1 unit 1`] = `
Array [
  "px-1",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 1/2 unit 1`] = `
Array [
  "px-half",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 1/4 unit 1`] = `
Array [
  "px-quarter",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 1/8 unit 1`] = `
Array [
  "px-eighth",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 2 unit 1`] = `
Array [
  "px-2",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 3 unit 1`] = `
Array [
  "px-3",
]
`;

exports[`padding px Generates the correct classes for padding x axis for the 4 unit 1`] = `
Array [
  "px-4",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 0 unit 1`] = `
Array [
  "py-0",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 1 unit 1`] = `
Array [
  "py-1",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 1/2 unit 1`] = `
Array [
  "py-half",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 1/4 unit 1`] = `
Array [
  "py-quarter",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 1/8 unit 1`] = `
Array [
  "py-eighth",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 2 unit 1`] = `
Array [
  "py-2",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 3 unit 1`] = `
Array [
  "py-3",
]
`;

exports[`padding py Generates the correct classes for padding y axis for the 4 unit 1`] = `
Array [
  "py-4",
]
`;
